Firestore/core/src/firebase/firestore/nanopb/writer.cc

```diff
@@ -72,20 +37,77 @@ void Writer::WriteNanopbMessage(const pb_field_t fields[],
   }
 }
 
-ByteStringWriter::ByteStringWriter() : Writer({}) {
-  stream_ = WrapContainer(&buffer_);
+static bool AppendToBytesArray(pb_ostream_t* stream,
+                               const pb_byte_t* buf,
+                               size_t count) {
+  auto writer = static_cast<ByteStringWriter*>(stream->state);
+  writer->Append(buf, count);
+  return true;
 }
 
-ByteString ByteStringWriter::ToByteString() const {
-  return ByteString(buffer_);
+ByteStringWriter::ByteStringWriter() : Writer() {
+  stream_.callback = AppendToBytesArray;
```

Comment:
`AppendToBytesArray` can avoid `extern "C"` because we compile nanopb in C++ mode?